File: ThisIsCodingTest/이것이코딩테스트다_11장_모험가길드_2.js
const N = 5;
const adventurers = [2, 3, 1, 2, 2];
adventurers.sort((a, b) => a - b);

let answer = 0;
let cnt = 0;

let i = 0;
while (i < N) {
    let j = i;

    while (j < N) {
        cnt += 1;

        if (adventurers[j] == cnt) {
            answer += 1;
            cnt = 0;
            break;
        } else {
            j += 1;
        }
    }

    i = j + 1;
}

console.log(answer);

File: this-is-coding-test/이것이코딩테스트다_11장_모험가길드_2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());

        int[] adventurers = new int[N];
        for (int i = 0; i < N; i++) {
            adventurers[i] = Integer.parseInt(st.nextToken());
        }
        int[] sortedAdventurers = Arrays
                .stream(adventurers)
                .sorted()
                .toArray();

        int total = 0;
        int cnt = 0;
        for (int a : sortedAdventurers) {
            cnt += 1;
            if (cnt == a) {
                total += 1;
                cnt = 0;
            }
        }

        System.out.println(total);
    }
}
